dll/ntdll/rtl/libsupp.c

```diff
@@ -1221,27 +1221,19 @@ ULONG
 NTAPI
 RtlGetTickCount(VOID)
 {
-    ULARGE_INTEGER TickCount;
+    LARGE_INTEGER TickCount;
+
+    TickCount = KiReadSystemTime(&SharedUserData->TickCount);
 
 #ifdef _WIN64
-    TickCount.QuadPart = *((volatile ULONG64*)&SharedUserData->TickCount);
+    return (TickCount.QuadPart * SharedUserData->TickCountMultiplier) >> 24;
```

Comment:
Can't you use the same code for both 32bits and 64bits? Does this have a different semantic depending on bitness?

Reply:
Semantically it should be the same. I'll have a look.

Reply:
I believe that a small performance improvement does not worth completely destroying the ability to understand what's happening here. The compiler supports 64bit math and all that  trickery with special macros or manually manipulating parts of integer should be left in 2005.

So my suggestion would be to just leave `(TickCount.QuadPart * SharedUserData->TickCountMultiplier) >> 24` here without any `#ifdef` and do that in other similar places.
Switching `LARGE_INTEGER` to `INT64` would be also nice where possible...

Reply:
I replaced it with an inline function.